common/src/stack/command/stack/commands/add/vm/__init__.py

```diff
@@ -60,15 +59,13 @@ def run(self, params, args):
 		if not self.is_hypervisor(hypervisor):
 			raise ParamError(self, param = 'hypervisor', msg = f'has a non valid hypervisor appliance or host')
 
-		# Cpu and Memory input must be valid numbers and be one or greater
-		# No guarantees your vm will boot with 1 MB of memory though
+		# CPU and Memory input must be valid numbers and be at least equal to one
 		if not cpu.isdigit() or int(cpu) < 1 or not memory.isdigit() or int(memory) < 1:
 			raise ParamError(self, 'cpu or memory', 'must be a number greater than 0')
 
 		# Check if the virtual machine is already defined
 		# Raise a CommandError if it is
-		for host in self.call('list.vm'):
-			if vm_host == host['virtual machine']:
+		if self.vm_by_name(vm_host):
 				raise CommandError(self, f'Virtual Machine {vm_host} already defined')
```

Comment:
Looks like the tabbing got a little out of wack here.

Reply:
Thanks for adding your input btw.